low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -501,10 +505,63 @@ def validate(self):
 
     def process(self):
         try:
+            args = ' '.join(self._args.args)
+            Conf.load(SSPL_CONFIG_INDEX, sspl_config_path)
+            sspl_log_file_path = Utility.get_config_value(
+                    SSPL_CONFIG_INDEX, "SYSTEM_INFORMATION>sspl_log_file_path")
+            iem_log_file_path = Utility.get_config_value(
+                    SSPL_CONFIG_INDEX, "IEMSENSOR>log_file_path")
             if os.path.exists(file_store_config_path):
                 os.remove(file_store_config_path)
             shutil.copyfile("%s/conf/sspl.conf.%s.yaml" % (
                 SSPL_BASE_DIR, self.product), file_store_config_path)
+
+            # --pre-factory will cleanup config and undo everything whatever
+            # we have done in post-install Mini-Provisioner Interface.
+            if '--pre-factory' in args:
```

Comment:
Also is this the way we have handled arguments everywhere else. if not, align to standard way

Reply:
@shriya-deshmukh, Explore `self.args`
dir(self.args) will show you the attributes received.